ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -130,21 +139,23 @@
    * @param config The store config used to initialize the index segment
    * @param metrics The store metrics used to track metrics
    * @param journal The journal to use
+   * @param time the {@link Time} instance to use
    * @throws StoreException
    */
   IndexSegment(File indexFile, boolean shouldMap, StoreKeyFactory factory, StoreConfig config, StoreMetrics metrics,
-      Journal journal) throws StoreException {
+      Journal journal, Time time) throws StoreException {
     try {
       startOffset = getIndexSegmentStartOffset(indexFile.getName());
       endOffset = new AtomicReference<>(startOffset);
       indexSegmentFilenamePrefix = generateIndexSegmentFilenamePrefix();
       this.indexFile = indexFile;
-      this.lastModifiedTimeSec = new AtomicLong(indexFile.lastModified() / 1000);
       this.rwLock = new ReentrantReadWriteLock();
       this.factory = factory;
+      this.time = time;
       sizeWritten = new AtomicLong(0);
       numberOfItems = new AtomicInteger(0);
       mapped = new AtomicBoolean(false);
+      lastModifiedTimeSec = new AtomicLong(0);
```

Comment:
Not sure I understand your comment. You can keep seconds inside `IndexSegment` - im just asking to introduce an ms function. `MessageInfo` and `IndexValue` historically have been in `ms`. This is the _only_ thing that is in seconds and that is because the file system exposes the last modified time in seconds.